extensions/panache/hibernate-orm-panache-rest/deployment/src/main/java/io/quarkus/hibernate/orm/panache/rest/deployment/utils/EntityUrlProvider.java

```diff
@@ -0,0 +1,41 @@
+package io.quarkus.hibernate.orm.panache.rest.deployment.utils;
+
+import java.net.URI;
+
+import io.quarkus.gizmo.BytecodeCreator;
+import io.quarkus.gizmo.MethodDescriptor;
+import io.quarkus.gizmo.ResultHandle;
+import io.quarkus.hibernate.orm.panache.rest.runtime.utils.StringUtil;
+
+public final class EntityUrlProvider {
+
+    private final EntityFieldAccessor fieldAccessor;
+
+    public EntityUrlProvider(EntityFieldAccessor fieldAccessor) {
+        this.fieldAccessor = fieldAccessor;
+    }
+
+    public static String getCollectionUrl(String entityType) {
+        return StringUtil.toPlural(StringUtil.camelToHyphenated(removePackage(entityType)));
+    }
+
+    private static String removePackage(String className) {
+        return className.substring(className.lastIndexOf(".") + 1);
+    }
+
+    public ResultHandle getEntityUrl(BytecodeCreator creator, ResultHandle entity, String entityType) {
```

Comment:
I think this should be provided by the user not automatically generated.
URL are the central point of the design of an API so a user must be able to manage them.

Reply:
We could allow a user to override a path segment (e.g. by default we use `/books` for Book.class) but we still need to generate a URL for each entity based on its ID (`/books/1`, `/books/2`...)

Reply:
We should support overriding this, but it can wait. We can make this an annotation on the controller like `@CRUD(singleUrl = "book", url = "books")`.

Reply:
Yes, that's what I was thinking. I just don't think that CRUD is a correct name if we plan to add more that these four methods. For now I've been using `@PanacheRestResource` annotation to define these things (so far `enable` and `hal`).